src/Data/PQueue/Prio/Max.hs

```diff
@@ -298,9 +305,24 @@ foldlWithKey f z0 (MaxPQ q) = Q.foldlWithKey (\z -> f z . unDown) z0 q
 -- (@'traverseWithKey' f q == 'fromDescList' <$> 'traverse' ('uncurry' f) ('toDescList' q)@)
 --
 -- If you do not care about the /order/ of the traversal, consider using 'traverseWithKeyU'.
+--
+-- If you are working in a strict monad, consider using 'mapMWithKey'.
 traverseWithKey :: (Ord k, Applicative f) => (k -> a -> f b) -> MaxPQueue k a -> f (MaxPQueue k b)
 traverseWithKey f (MaxPQ q) = MaxPQ <$> Q.traverseWithKey (f . unDown) q
 
+-- | A version of 'traverseWithKey' designed to be efficient for strict monads
+-- such as 'IO' and strict @State@.
+mapMWithKey :: (Ord k, Monad m) => (k -> a -> m b) -> MaxPQueue k a -> m (MaxPQueue k b)
+mapMWithKey f = go empty
+  where
+    go acc q = acc `seq`
```

Comment:
```suggestion
    go !acc q =
```
If we already enable `BangPatterns`, we might as well use them.

Reply:
Yes, true. There are a *couple* spots in the code where I'd prefer to keep the `seq` explicit (forcing on cascade), but this isn't one of them.